zipline/assets/asset_writer.py

```diff
@@ -563,3 +560,9 @@ def _load_data(self, equities, futures, exchanges, root_symbols):
             exchanges=exchanges_output,
             root_symbols=root_symbols_output,
         )
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *exc_info):
+        pass  # Shouldn't dispose of the engine explicitly
```

Comment:
Can you explain more why we don't need to dispose of the engine? I afree that we should either undo this change, or make it actually dispose of the engine and update the callers to use the ctx manager aspect of the object.